boat-maven-plugin/src/main/java/com/backbase/oss/boat/RadioMojo.java

```diff
@@ -0,0 +1,171 @@
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+
+@Mojo(name = "radio", requiresDependencyResolution = ResolutionScope.RUNTIME, threadSafe = true)
+@Slf4j
+@Getter
+@Setter
+public class RadioMojo extends AbstractMojo {
+
+    @Parameter(property = "groupId", defaultValue = "${project.groupId}", required = true)
+    private String groupId;
+
+    @Parameter(property = "artifactId", defaultValue = "${project.artifactId}", required = true)
+    private String artifactId;
+
+    @Parameter(property = "version", defaultValue = "${project.version}", required = true)
+    private String version;
+
+    @Parameter(property = "basePath", required = true)
+    private String basePath;
+
+    @Parameter(property = "portalKey", required = true)
+    private String portalKey;
+
+    @Parameter(property = "sourceKey", required = true)
+    private String sourceKey;
+
+    @Parameter(property = "username", required = false)
+    private String username;
+
+    @Parameter(property = "password", required = false)
+    private String password;
+
+    @Parameter(property = "specs")
+    private SpecConfig[] specs;
+
+    @Parameter(name = "radioOutput", defaultValue = "${project.build.directory}/target/boat-radio-report")
+    private File radioOutput;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+
+        BasicAuthRequestInterceptor basicAuthRequestInterceptor = null;
+
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        if(username != null && !username.isEmpty() && password != null && !password.isEmpty()){
+             basicAuthRequestInterceptor = new BasicAuthRequestInterceptor(username, password);
+        }else{
+            getLog().warn("No Authentication set");
+        }
+
+        List<UploadSpec> allSpecs = Arrays.stream(specs).map(this::mapToUploadSpec).collect(Collectors.toList());
+
+        ApiClient apiClient = new ApiClient().setBasePath(basePath);
+        if(basicAuthRequestInterceptor != null){
+            apiClient.addAuthorization("Basic Auth", basicAuthRequestInterceptor);
+        }
+
+        BoatMavenPluginApi api = apiClient.buildClient(BoatMavenPluginApi.class);
+
+        UploadRequestBody uploadRequestBody = new UploadRequestBody();
+        uploadRequestBody.setGroupId(groupId);
+        uploadRequestBody.setArtifactId(artifactId);
+        uploadRequestBody.setVersion(version);
+        uploadRequestBody.setSpecs(allSpecs);
+
+        List<BoatLintReport> reports = api.uploadSpec(portalKey,sourceKey,uploadRequestBody);
+
+        try {
+            objectMapper.writerWithDefaultPrettyPrinter().writeValue(new File(getOutput(), "radioOutput.json"),reports);
+        } catch (IOException e) {
+            throw new MojoFailureException("Failed to write output", e);
```

Comment:
Do we get a log of what the absolute output file is?

Reply:
We should also print the link to the report in the output of the maven plugin like you get when you are running mvn sonar:sonar

Reply:
Done.
At the end it prints something like 

```
[info] Changes to spec spec-key is COMPATIBLE
[info] UPLOAD TO BOAT-BAY SUCCESSFUL, check the full report: /Users/dhananjay/Documents/BB/GITHUB/BOAT/backbase-openapi-tools/boat-maven-plugin/target/boat-radio-report/radioOutput.json
```